Support for ili9488 displays

wightmanjr opened the pull request:
Updated const.py and device.py to support ILI9488 displays.

thijstriemstra commented:
thanks! couple of things:

- can you your name to CONTRIBUTING.rst?
- can you take a look at adding tests? 

let us know if you have any questions.

wightmanjr commented:
I will work on it. I'm somewhat new to python tests, but will figure it out as I go.

Thanks

thijstriemstra commented:
thanks. also see https://github.com/rm-hull/luma.lcd/actions/runs/5232388084/jobs/9451907761#step:9:24 for some linter failures causing the build to fail.

wightmanjr commented:
- Added name to CONTRIBUTING.rst

- Updated documentation

- Uploaded ILI9488 datasheet

- Fixed flake errors

- Updated gamma values

- Added tests for ILI9488

thijstriemstra commented on doc/hardware.rst:
```diff
@@ -335,6 +335,26 @@ SCK or CLK    SPI clock         P01-23   GPIO 11 (SCLK)
+~~~~~~~
+No support for the touch-screen, leave the MISO and Touch pins disconnected.
+LED(Backlight) draws around 120ma, you may need to a Current Driver to avoid
+overloading the IO pin.Depending on the board you bought, there may be 
```
space between `pin.` and `Depending` is missing.

thijstriemstra commented on luma/lcd/device.py:
```diff
@@ -879,6 +879,131 @@ def command(self, cmd, *args):
+
+        # Initialization sequence, adapted from
+        #
+        # `https://github.com/birdtechstep/fbtft/blob/master/fb_ili9488.c`
```
no need for backticks around the url here

wightmanjr replied:
Removed backticks

thijstriemstra replied:
@wightmanjr the branch doesn't show any new commits though..

thijstriemstra reviewed (changes requested):
thanks! some minor things left.

rm-hull commented:
Aside @thijstriemstra's comments, looks fine to me

wightmanjr commented on doc/hardware.rst:
```diff
@@ -335,6 +335,26 @@ SCK or CLK    SPI clock         P01-23   GPIO 11 (SCLK)
+ILI9488
+~~~~~~~
+No support for the touch-screen, leave the MISO and Touch pins disconnected.
+LED(Backlight) draws around 120ma, you may need to a Current Driver to avoid
```
fixed "add" and "current driver"

wightmanjr commented on luma/lcd/device.py:
```diff
@@ -879,6 +879,131 @@ def command(self, cmd, *args):
+            self.apply_offsets = lambda bbox: bbox
+
+        # Supported modes
+        supported = (width, height) in [(480, 320)]  # full
```
removed full

wightmanjr commented:
Made changes @thijstriemstra requested.

thijstriemstra commented:
@wightmanjr there are no new commits since my last review?

thijstriemstra reviewed (approved):
thanks!

wightmanjr commented:
Thanks for all the help, it was a great learning experience.
